### PR TITLE
Simple script to allocate & assign storage for compute nodes

### PR DESCRIPTION
**Pull request by NateThornton**
```
usage: allocate-compute.py [-h] [--node {0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15} [{0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15} ...]] [--size SIZE] [--host HOST] [--port PORT]

Allocate storage and assign it to a compute node(s)

optional arguments:
  -h, --help            show this help message and exit
  --node {0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15} [{0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15} ...]
                        the node index to attach to
  --size SIZE           storage pool size (default 500GB)
  --host HOST           specify a server endpoint
  --port PORT           specify a server port

Must executed on Rabbit with nnf-ec running
```

Signed-off-by: Nate Thornton <nate.thornton@hpe.com>

### CODE REVIEW

File: scripts/allocate-compute.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import argparse, sys, time
+
+import common, connection
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description='Allocate storage and assign it to a coumpte node',
+        epilog = 'Must executed on Rabbit with nnf-ec running')
+    
+    parser.add_argument('--node', nargs='+', choices=range(0,16), default=None, help='the node index to attach to')
```
Comment (ajfloeder) [inline scripts/allocate-compute.py:12]: How would a user know that 'None' means 'All'? Wouldn't the default be 'all'?

File: scripts/allocate-compute.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+
+import argparse, sys, time
+
+import common, connection
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description='Allocate storage and assign it to a coumpte node',
```
Comment (bdevcich-hpe) [inline scripts/allocate-compute.py:9]:
```suggestion
        description='Allocate storage and assign it to a compute node',
```

File: scripts/allocate-compute.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    args = parser.parse_args()
+
+    if args.node is None:
+        args.node = range(0, 16)
+
+    conn = connection.connect(args, '/redfish/v1/StorageServices/NNF')
+
+    for node in args.node:
+        print(f"Starting Node '{node}' Sequence...")
```
Comment (bdevcich-hpe) [inline scripts/allocate-compute.py:25]: For the prints, you're using double quotes in some places and single quotes elsewhere.

File: scripts/allocate-compute.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    parser.add_argument('--size', type=str, action=common.ByteSizeAction, default=common.ByteSizeStringToIntegerBytes('500GB'), help='storage pool size (default 500GB)')
+
+    connection.addServerArguments(parser)
+
+    args = parser.parse_args()
+
+    if args.node is None:
+        args.node = range(0, 16)
+
+    conn = connection.connect(args, '/redfish/v1/StorageServices/NNF')
```
Comment (bdevcich-hpe) [inline scripts/allocate-compute.py:22]: Handle error here?

File: scripts/allocate-compute.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+cm power status -t node x9000c3s*b*n* | awk -F :  '{print $1}'
+readarray SYSTEMS < <(cm power status -t node x9000c3s*b*n* | awk -F :  '{print $1}')
+
+for SYSTEM in ${SYSTEMS[@]}; do ssh $SYSTEM 'for DRIVE in $(ls -v /dev/nvme* | grep -E "nvme[[:digit:]]+n[[:digit:]]+$"); do nvme id-ns $DRIVE | grep "NVME"; done'; done
+
+SYSTEM=${SYSTEMS[0]}
+ssh $SYSTEM 'for DRIVE in $(ls -v /dev/nvme* | grep -E "nvme[[:digit:]]+n[[:digit:]]+$"); do nvme id-ns $DRIVE | grep "NVME"; done'
+ssh $SYSTEM ls /dev/rabbit/rabbit
+
+"""
```
Comment (bdevcich-hpe) [inline scripts/allocate-compute.py:84]: What is this last block for?

Comment (bdevcich-hpe) [inline scripts/allocate-compute.py:84]: Missing newline at end of file too

**Review by bdevcich-hpe — COMMENTED** [review]
Since this is a brand new file, I recommend running this through `black` or something.

File: scripts/allocate-compute.py (hunk 3 above)
Comment (NateThornton) [reply scripts/allocate-compute.py:25]: Double only when string uses single quotes

File: scripts/allocate-compute.py (hunk 4 above)
Comment (NateThornton) [reply scripts/allocate-compute.py:22]: I prefer the crash

File: scripts/allocate-compute.py (hunk 5 above)
Comment (NateThornton) [reply scripts/allocate-compute.py:84]: notes - removed

File: scripts/allocate-compute.py (hunk 1 above)
Comment (NateThornton) [reply scripts/allocate-compute.py:12]: I updated the usage

**Review by bdevcich-hpe — APPROVED** [review]

**Review by ajfloeder — APPROVED** [review]

**Pull request merged**